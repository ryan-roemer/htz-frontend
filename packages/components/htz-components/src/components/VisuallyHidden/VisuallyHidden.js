import React from 'react';
import { createComponent, } from 'react-fela';
import PropTypes from 'prop-types';
import { visuallyHidden, } from '@haaretz/htz-css-tools';

const VisuallyHiddenStyled = createComponent(visuallyHidden, 'span', [ 'id', ]);

VisuallyHidden.propTypes = {
  /** ID for the HTML element (to be referenced with the aria-describedby attribute) */
  id: PropTypes.string,
  children: PropTypes.node,
};

VisuallyHidden.defaultProps = {
  id: null,
  children: null,
};

function VisuallyHidden({ id, children, }) {
  return <VisuallyHiddenStyled {...id ? { id, } : {}}>{children}</VisuallyHiddenStyled>;
}
export default VisuallyHidden;
