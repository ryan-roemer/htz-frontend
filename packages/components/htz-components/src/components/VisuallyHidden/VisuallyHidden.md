Creates an invisible span tag, that is readable by screen-readers. <br/>
This is used for Accessibility.

```html
<p aria-describedby="info">
    This is visible to human eyes.
    <span id="info" class="visually-hidden-class-name">This text is invisible</span>  
</p>
```

```jsx
<div
  aria-describedby="my-div-description"
  style={{ border: "1px green dashed" }}
>
  This div element is described by an invisble element.<br />
  Look at page-source to see div description.
  <VisuallyHidden id="my-div-description">
    This text describes the container
  </VisuallyHidden>
</div>
```
