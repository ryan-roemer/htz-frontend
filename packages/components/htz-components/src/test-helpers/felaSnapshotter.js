import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
