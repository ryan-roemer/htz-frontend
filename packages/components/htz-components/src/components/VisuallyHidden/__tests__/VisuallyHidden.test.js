import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import VisuallyHidden from '../VisuallyHidden';

describe('VisuallyHidden component', () => {
  it('renders correctly', () => {
    const snapshot = felaSnapshotter(
      <VisuallyHidden id="my-div-description">
        This text describes the container
      </VisuallyHidden>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
