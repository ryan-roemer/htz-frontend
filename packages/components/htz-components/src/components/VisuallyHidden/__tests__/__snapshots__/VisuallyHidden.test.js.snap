// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisuallyHidden component renders correctly 1`] = `
Object {
  "component": <div>
    <span
      className="htz-a htz-b htz-c htz-d htz-e htz-f htz-g htz-h htz-i htz-j htz-k"
      id="my-div-description"
    >
      This text describes the container
    </span>
  </div>,
  "styles": ".htz-a{border:0}.htz-b{clip:rect(0 0 0 0)}.htz-c{-webkit-clip-path:inset(50%)}.htz-d{clip-path:inset(50%)}.htz-e{height:1px}.htz-f{margin:-1px}.htz-g{overflow:hidden}.htz-h{padding:0}.htz-i{position:absolute}.htz-j{width:1px}.htz-k{white-space:nowrap}",
}
`;
